### PR TITLE
JENKINS-60924 Add PostCSS pre-processor

### PR DESCRIPTION
**Pull request by 95jonpet**
See [JENKINS-60924](https://issues.jenkins-ci.org/browse/JENKINS-60924).

Adds [PostCSS](https://postcss.org/) to the [Webpack](https://webpack.js.org/) build process. Also adds the [autoprefixer](https://github.com/postcss/autoprefixer) plugin that automatically inserts vendor prefixes in the outputted CSS.

### Proposed changelog entries

* Internal: Add PostCSS pre-processor for CSS and automatically add browser vendor prefixes.

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@fqueiruga 

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Review by timja — COMMENTED** [review]
Do any styles need removing due to adding this?

Btw I've added https://issues.jenkins-ci.org/browse/JENKINS-62473 if you feel like tackling that one too

**Review by fqueiruga — APPROVED** [review]

**Review by timja — APPROVED** [review]

**Comment by fqueiruga** [comment]
Tested it and the bundles are definitely bigger, which means prefixes are being added.

I don't think we should remove anything now, we can do it progressively. We can look for `-ms`, `-moz` prefixes on less files for example.

**Review by oleg-nenashev — APPROVED** [review]
We discussed it at the UI/UX Hackfest meeting, and the consensus is that we need to merge it first to unblock the CSS variables which we need for a Dark Theme. Full notes: https://docs.google.com/document/d/1Wve6fXV7mPkqtFn_gR9kfDWq-qOu6pPLYcDzKTjOHoI/edit#bookmark=id.2n9shyr9hg3n

At the meeting we agreed that it is a low-risk change. We will be testing it today with our prototypes, but I will start the merge countdown taking the approvals.

We may merge it in 24 hours if no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information

**Comment by 95jonpet** [comment]
Already existing vendor prefixes could be cleaned up in the source files, but I think that it is a separate issue.

File: war/postcss.config.js
Comment (halkeye) [inline war/postcss.config.js:4]: So I think by default autoprefixer uses the browser list from package.json (or other files), and then if not defined, does its own list of browsers.

I think we need to make sure we have a browserlist (https://www.npmjs.com/package/browserslist) defined that matches the browser compatibility matrix that matrix jenkins - https://www.jenkins.io/doc/administration/requirements/web-browsers/ (since its a relative list, its pretty easy to define in browserlist)

Comment (fqueiruga) [reply war/postcss.config.js:4]: There's one on package.json https://github.com/jenkinsci/jenkins/blob/f5fdc4bc4c0e11cc47f21475cd137d8eda2ba3e2/war/package.json#L54

Comment (halkeye) [reply war/postcss.config.js:4]: oh cool, i totally missed that when i scaned

👍 👍

**Review by halkeye — APPROVED** [review]

**Comment by oleg-nenashev** [comment]
> Already existing vendor prefixes could be cleaned up in the source files, but I think that it is a separate issue.

Would be great if somebody could report a follow-up issue for it

**Comment by halkeye** [comment]
> Would be great if somebody could report a follow-up issue for it

worth looking into stylelint too, which could probably auto fix / report vendor prefixes

**Pull request merged**
